[xSAN]Explicitly link against libresolv for Sanitizer IB

smuzaffar opened the pull request:
This should fix the ASAN/UBSAN runtime errors [a]. See https://github.com/llvm/llvm-project/issues/59007#issuecomment-2474131218 for details. 
```
The specific issue should only occur with older glibc versions: the call to dn_expand [was removed](https://sourceware.org/git/?p=glibc.git;a=commitdiff;h=75b3a15e077dbfdfd8cbb3449369379e700b9972) from getanswer_r in glibc 2.26. With version 2.34, dn_expand [has been moved](https://sourceware.org/git/?p=glibc.git;a=commitdiff;h=640bbdf71c6f10ac26252ac67a22902e26657bd8) from libresolv.so to libc.so. However, I think we should still add a link to -lresolv, if only because that's where the functions are documented to sit. And my understanding is that it's a no-op on newer systems anyway.
```

linking frontier_client library with`libresolv` does not fix the issue as looks like `libresolv` should be loaded directly by the application (e.g. `cmsRun`) and not via EdmPluginManager (dlopen)

cmsbuild commented:
A new Pull Request was created by  @smuzaffar for master.

It involves the following packages:

- FWCore/Utilities (**core**)


@Dr15Jones, @cmsbuild, @makortel, @smuzaffar can you please review it and eventually sign? Thanks.
@felicepantaleo, @makortel, @missirol, @wddgit this is something you requested to watch as well.
@antoniovilela, @mandrenguyen, @rappoccio, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

smuzaffar commented:
please test for CMSSW_15_1_X/el8_amd64_gcc13 using full cmssw

lets see if this changes fixes all unit tests

cmsbuild commented:
+1

**Size**: This PR adds an extra 20KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2ae5de/46372/summary.html
**COMMIT**: 2835dbcd51fdcefd5424e7f75584290b0235b36d
**CMSSW**: CMSSW_15_1_X_2025-05-26-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48181/46372/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 1 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-05-26-2300+2ae5de/68704/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 6 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-05-26-2300+2ae5de/68704/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4038193
 - DQMHistoTests: [Total failures: 6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-05-26-2300+2ae5de/68704/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4038167
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-05-26-2300+2ae5de/68704/triggerResults): no differences found

smuzaffar commented:
please test for CMSSW_15_1_ASAN_X/el8_amd64_gcc13

smuzaffar commented:
> Ok, if it has to be loaded that early, I'd (also) prefer to add the dependence directly to cmsRun (along with an explanatory comment).

@makortel , I have updated this PR and added the dependency for `cmsRun` only. Currently I added it only for default `cmsRun`, let me know if this should be added for all `cmsRunXX` (in that case I can just move the newly added block out side of  ` <bin name="cmsRun" file="cmsRun.cpp"> ... </bin>`) ?

smuzaffar committed:
[xSAN]Explicitly link against libresolv for el8/gcc13|4 Sanitizer IB

cmsbuild commented:
Pull request #48181 was updated. @Dr15Jones, @makortel, @smuzaffar can you please check and sign again.

makortel commented:
> > Ok, if it has to be loaded that early, I'd (also) prefer to add the dependence directly to cmsRun (along with an explanatory comment).
> 
> @makortel , I have updated this PR and added the dependency for `cmsRun` only. Currently I added it only for default `cmsRun`, let me know if this should be added for all `cmsRunXX` (in that case I can just move the newly added block out side of ` <bin name="cmsRun" file="cmsRun.cpp"> ... </bin>`) ?

Hmm. I suppose the other `cmsRun` variants would fail to run in xSAN builds. I'd guess the probability for anyone to run `cmsRunXX` in xSAN builds is low, but maybe nevertheless it would be better to try to keep them functional.

smuzaffar committed:
explicitly link libresolv for all variants of cmsRun

smuzaffar commented:
please test

cmsbuild commented:
Pull request #48181 was updated. @Dr15Jones, @makortel, @smuzaffar can you please check and sign again.

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2ae5de/46447/summary.html
**COMMIT**: 7cf4066bcaecef054dd56e5cda25290c54048861
**CMSSW**: CMSSW_15_1_ASAN_X_2025-05-28-2300/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48181/46447/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

The relvals timed out after 4 hours.
- [29834.999](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2ae5de/46447/runTheMatrix-results/29834.999_TTbar_14TeV+Run4D110PU_PMXS1S2PR)```29834.999_TTbar_14TeV+Run4D110PU_PMXS1S2PR/step4_TTbar_14TeV+Run4D110PU_PMXS1S2PR.log```
- [29634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2ae5de/46447/runTheMatrix-results/29634.0_TTbar_14TeV+Run4D110)```29634.0_TTbar_14TeV+Run4D110/step3_TTbar_14TeV+Run4D110.log```
- [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2ae5de/46447/runTheMatrix-results/24834.911_TTbar_14TeV+Run4D98_DD4hep)```24834.911_TTbar_14TeV+Run4D98_DD4hep/step3_TTbar_14TeV+Run4D98_DD4hep.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [29634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2ae5de/46447/runTheMatrix-results/29634.911_TTbar_14TeV+Run4D110_DD4hep)
- [145.301](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2ae5de/46447/runTheMatrix-results/145.301_RunDisplacedJet2024E)
</details>

smuzaffar commented:
+core

technical update
